pkg/ovs/utils.go

```diff
@@ -61,6 +63,28 @@ func getPhysicalNetworks(
 	)
 }
 
+func getOvsPodsNumber(
+	ctx context.Context,
+	k8sClient client.Client,
+	instance *v1beta1.OVS,
+) (int32, error) {
+
+	podList := &corev1.PodList{}
+	podListOpts := &client.ListOptions{
+		Namespace: instance.Namespace,
+	}
+	client.MatchingLabels{
+		"service": ServiceName,
+	}.ApplyToList(podListOpts)
+
+	if err := k8sClient.List(ctx, podList, podListOpts); err != nil {
+		err = fmt.Errorf("error listing pods for instance %s: %w", instance.Name, err)
+		return 0, err
+	}
+
+	return int32(len(podList.Items)), nil
+}
```

Comment:
Something like:

```
func getOvsPodNodes(
	ctx context.Context,
	k8sClient client.Client,
	instance *v1beta1.OVS,
) ([]string, error) {
    var nodes []string
	podList := &corev1.PodList{}
	podListOpts := &client.ListOptions{
		Namespace: instance.Namespace,
	}
	client.MatchingLabels{
		"service": ServiceName,
	}.ApplyToList(podListOpts)

	if err := k8sClient.List(ctx, podList, podListOpts); err != nil {
		err = fmt.Errorf("error listing pods for instance %s: %w", instance.Name, err)
		return []string{}, err
	}

	for _, pod := range podList.Items {
		nodes = append(nodes, pod.Spec.NodeName)
	}

	return nodes, nil
}
```

Then we can use the list of nodes, and we could set `NodeName` rather than a `nodeSelector`. Which would ensure we definitely run on the correct Node. We would use need to iterate over the list and create one job per node though.

Reply:
Thx, yes, I was thinking about something like that but my initial idea was to make it simpler. But it's true that I didn't consider the case when 2 pods with config job will run on the same node. I will change it.